proto/substrait/algebra.proto

```diff
@@ -726,6 +730,52 @@ message NestedLoopJoinRel {
   substrait.extensions.AdvancedExtension advanced_extension = 10;
 }
 
+message DelimGetRel{
+   RelCommon common = 1;
+   ReferenceRel input = 2;
+   repeated Expression.FieldReference column_ids = 3;
+}
+
+message DelimJoinRel {
+  RelCommon common = 1;
+  Rel left = 2;
+  Rel right = 3;
+
+  Expression expression = 4;
+  Expression post_join_filter = 5;
+
+  JoinType type = 6;
+
+  // The set of columns that will be duplicate eliminated from the LHS and pushed into the RHS
+  repeated Expression.FieldReference duplicate_eliminated_columns = 7;
+
+  DelimiterSide delimiter_side = 8;
+
+  // If this is a DelimJoin, whether it has been flipped to de-duplicating the LHS or RHS
+  enum DelimiterSide {
+      LEFT = 0;
```

Comment:
Based on the check protobuf action it looks like this should be:

DELIMITER_SIDE_UNSPECIFIED = 0;
DELIMITER_SIDE_LEFT = 1;
DELIMITER_SIDE_RIGHT = 2;

But given that this feature is actually duplicate eliminated joins should we be calling it delimiter here or something else?

Reply:
Absolutely, maybe we should go for DuplicateEliminatedSide?
I think I'm never going to be able to get delimiter out of my head 😂

Reply:
Works for me.